clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3148,13 +3148,13 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
   // to the default value 1, but only if the sycl:: or intel::
   // reqd_work_group_size spelling was used.
   auto SetDefaultValue = [](Sema &S, const ParsedAttr &AL, SourceLocation loc) {
-    Expr *E =
-        (AL.getKind() == ParsedAttr::AT_ReqdWorkGroupSize && AL.hasScope() &&
-         (AL.getScopeName()->isStr("sycl") ||
-          AL.getScopeName()->isStr("intel")))
-            ? IntegerLiteral::Create(S.Context, llvm::APInt(32, 1),
-                                     S.Context.IntTy, AL.getLoc())
-            : nullptr;
+    assert((AL.getKind() == ParsedAttr::AT_ReqdWorkGroupSize && AL.hasScope() &&
+            (AL.getScopeName()->isStr("sycl") ||
+             AL.getScopeName()->isStr("intel"))) &&
+           "Attribute does not exist in sycl:: or intel:: scope");
+
+    Expr *E = IntegerLiteral::Create(S.Context, llvm::APInt(32, 1),
```

Comment:
Rather than create the literal, storing it into `E`, and then returning `E`, just return the literal directly (no local variable).